build/cmake/SGX.cmake

```diff
@@ -28,8 +28,18 @@ SET(SGX_MODE $ENV{SGX_MODE})
 
 IF (${SGX_MODE} STREQUAL "SIM")
     SET(SGX_USE_SIMULATOR TRUE)
+
+    IF (${PDO_DEBUG_BUILD} STREQUAL "0")
+        MESSAGE(FATAL_ERROR "SGX_MODE=SIM does not accept PDO_DEBUG_BUILD=0")
+    ENDIF()
```

Comment:
goes back to my earlier question about definition of PDO_DEBUG_BUILD: what does "SGX_MODE=SIM does not accept PDO_DEBUG_BUILD=0" mean ?

Reply:
It means that PDO_DEBUG_BUILD must be 1 in SIM mode. As SIM mode is by definition not a production mode, it assumed that debug must be turned on.

Reply:
Given that we are trying to hide `PDO_DEBUG_BUILD` behind `CMAKE_BUILD_TYPE` wouldn't it be good to also mention the latter as we do below for `abort if build_type=release and PDO_DEBUG_BUILD=1`?